### PR TITLE
[ROCm][CI] Gate incompatible HF references on Transformers v5

### PR DESCRIPTION
**Pull request by AndreasKaratzas**
This updates generation-test metadata for models whose HF reference path is incompatible with the installed Transformers v5 runtime.

Test groups fixed:
- `mi355_1: Language Models Tests (Standard)` for MiniCPM4.
- `mi355_1: Language Models Test (Extended Generation)` for HyperCLOVAX.

MiniCPM4:
- Fails in HF remote code because it imports `is_torch_fx_available`, removed from the installed Transformers v5 runtime.
- Links the upstream compatibility request, which is closed as not planned: https://github.com/huggingface/transformers/issues/44561

HyperCLOVAX:
- The model's remote code imports `ROPE_INIT_FUNCTIONS` and unconditionally indexes `ROPE_INIT_FUNCTIONS[self.rope_type]`; for the default case, `self.rope_type` is `"default"`: https://huggingface.co/naver-hyperclovax/HyperCLOVAX-SEED-Think-14B/blob/main/modeling_hyperclovax.py
- Transformers v5 docs still list `"default"` as a valid RoPE type. The incompatibility is narrower: v5 no longer exposes a `"default"` entry in `ROPE_INIT_FUNCTIONS`, while v4.57 did. In v5's own Llama code, default RoPE is handled by `compute_default_rope_parameters`; `ROPE_INIT_FUNCTIONS` is consulted only when `rope_type != "default"`.
- The v5 migration guide also notes that rotary embedding configuration now lives under `config.rope_parameters`, which matches the new v5 model path.

cc @kenroche

### CODE REVIEW

File: tests/models/registry.py [hunk 1]
```diff
@@ -335,6 +335,12 @@ def check_available_online(
     "HYV3ForCausalLM": _HfExamplesInfo("tencent/Hy3-preview", trust_remote_code=True),
     "HyperCLOVAXForCausalLM": _HfExamplesInfo(
         "naver-hyperclovax/HyperCLOVAX-SEED-Think-14B",
+        max_transformers_version="4.57",
+        transformers_version_reason={
+            "hf": "HF remote code indexes ROPE_INIT_FUNCTIONS['default']; "
+            "Transformers v5 supports default RoPE but handles it outside "
+            "ROPE_INIT_FUNCTIONS."
+        },
```
[the remark this answers is not in the document]
Comment (AndreasKaratzas) [reply tests/models/registry.py]: Fixed in my latest commit.

File: tests/models/registry.py [hunk 2]
```diff
@@ -419,7 +425,15 @@ def check_available_online(
         "openbmb/MiniCPM3-4B", trust_remote_code=True
     ),
     "MiniCPM4ForCausalLM": _HfExamplesInfo(
-        "openbmb/MiniCPM4.1-8B", trust_remote_code=True
+        "openbmb/MiniCPM4.1-8B",
+        min_transformers_version="4.56",
+        max_transformers_version="4.57",
```
[the remark this answers is not in the document]
Comment (AndreasKaratzas) [reply tests/models/registry.py:429]: The 4.56 to 4.57 range is intentional for MiniCPM4.1. This repo requires transformers >= 4.56.0, and current Transformers is v5. The max cap is also intentional: the remote code imports is_torch_fx_available, which was removed in Transformers v5.0, and the upstream compatibility shim request was closed as not planned: 
https://github.com/huggingface/transformers/issues/44561

**Comment by AndreasKaratzas** [comment]
cc @charlifu

**Comment by AndreasKaratzas** [comment]
Notes on the AudioFlamingo3 / MusicFlamingo changes:
- Transformers 5.5 native processors now require a 1:1 mapping between `text` entries and `audio` entries. vLLM's processor path supports multiple audio items inside one prompt, so the AudioFlamingo3 multimodal processor now performs the HF processor's windowing, feature extraction, prompt expansion, and tokenization locally for this single-prompt multi-audio case.
- MusicFlamingo's native processor no longer returns `rote_timestamps`. vLLM now derives the same rotary-time timestamps from `chunk_counts` and the encoded audio sequence length inside the model path, while still accepting `rote_timestamps` if an older processor provides them.
- MusicFlamingo prompt expansion now includes the audio BOS/EOS tokens expected by its prompt replacement metadata.
- `transformers_version_reason` now only applies when the installed version misses the configured min/max bound. This keeps the AudioFlamingo3/MusicFlamingo "Needs PR 43538" explanation for versions below `5.3.0` without skipping supported `5.5.3` runs.

**Review by hmellor — DISMISSED** [review]
MusicFlamingo is supported in Transformers 5.5, it was added in https://github.com/huggingface/transformers/commit/a9c6700a5078e8a9276656a0d0b82b32958624b7, it should not be skipped.

**Review by eustlb — COMMENTED** [review]
`transformers` releases
AudioFlamingo3: v5.0.0
MusicFlamingo3: v5.5.0

There should not be any reasons not to use `super()._call_hf_processor`. Looks like a lot of the confusion here comes from the fact that the vLLM PR was merged before the `transformers` one on which it depended, and that evolved after the vLLM one was merged.

Such fixes should be addressed in work already started in #39011, but happy to help if needed. @lashahub looping you in here

**Comment by AndreasKaratzas** [comment]
I pushed some more changes (a lot of them actually). The main intent is to stop carrying vLLM-local copies of preprocessing that now belongs to the upstream HF processors. So vLLM had manual chunking, feature extraction, and audio-token expansion logic. With AudioFlamingo3 at transformers `v5.0.0` and MusicFlamingo at `v5.5.0`, that logic should live in HF, while vLLM keeps only the adapter bookkeeping needed for batching, placeholder matching, and model execution.

In `audioflamingo3.py`, `_call_hf_processor` now calls `super()._call_hf_processor(...)`. The deleted active-path code was duplicating HF behavior: manually splitting audio into chunks, calling the feature extractor directly, normalizing masks, expanding `<sound>` tokens, then calling the tokenizer. Current HF `AudioFlamingo3Processor.__call__` already does those pieces.

The remaining vLLM logic after `super()` is:

- Copy `mm_data` first so we do not mutate the caller's mapping with `pop`.
- Map vLLM's `"audios"` alias to HF's `"audio"` argument.
- Rename HF's `input_features_mask` to vLLM's `feature_attention_mask`.
- Compute `chunk_counts`, because HF returns flattened audio chunks but vLLM still needs to regroup chunk embeddings back per original audio item.

So `chunk_counts` is not a replacement for HF preprocessing. It is vLLM batching metadata.

I also removed MusicFlamingo's `_call_hf_processor` because after the AudioFlamingo3 parent method was fixed, the MusicFlamingo override only duplicated parent behavior and recomputed `chunk_counts`. MusicFlamingo can inherit the shared AudioFlamingo3 adapter path.

Also regarding MusicFlamingo's `_expand_audio_tokens`, I think that current HF `MusicFlamingoProcessor` expands audio placeholders itself, including the MusicFlamingo-specific BOS/audio/EOS pattern. Keeping a second vLLM implementation risks drifting from HF again. vLLM still keeps MusicFlamingo prompt replacement in `musicflamingo.py`, because vLLM needs to match generated placeholder token IDs to multimodal embedding slots.

The `partial_rotary_factor` change in `musicflamingo.py` aligns with current transformers. The previous code used the full head dimension. HF computes `dim = int(head_dim * partial_rotary_factor)`, so I think that vLLM was rotating the wrong dimensionality.

The RoPE forward change from flattened `batch_positions` to timestamp-derived `window_positions` also matches current HF. The old implementation made the rotary window axis depend on flattened chunk order. Current HF uses the actual window start timestamp, computes the audio window duration, then derives the window position from that. That matters for chunked audio and for matching single vs batched behavior.

The fp32 RoPE buffer restoration is there because vLLM model dtype/device application can cast non-persistent buffers to bf16. In the HF construction path we compare against, `inv_freq` and `position_angles` are built from `torch.float`, so they remain fp32 under the tested bf16 load path. These buffers feed trig functions; bf16 there is enough to create exact-output drift. `_apply` lets the module move to the right device, then recreates those RoPE buffers in fp32 on that device.

The cast in `_encode_audio_features` fixes the ROCm failure where processor output is fp32 but the audio tower conv weights/bias are bf16. ROCm/PyTorch requires input and bias dtypes to match for that conv path. Casting at model ingress is the right boundary: HF processors naturally produce float features, while the vLLM model owns execution dtype/device.

Also, for ROCm, the transcription was wrong yielding a weird "four-on-the-floor" with MusicFlamingo. Direct HF generation matched the original fixture, while vLLM's URL audio path produced a different transcription. The root cause was audio resampling: HF/librosa uses soxr-style resampling, while vLLM was using the default parser resampler. Switching only MusicFlamingo's parser to `audio_resample_method="soxr"` makes the input features match HF more closely without changing the global default for other models.

The MusicFlamingo generation test no longer skips missing fixtures because missing committed fixtures should be a test error, not a skip. The small warmup in `test_musicflamingo.py` is there because MI300/ROCm showed a first-inference exact-output drift during decoder kernel compilation; after a one-token warmup, the steady-state deterministic output matches.

cc @eustlb @hmellor Lmk what you guys think.

**Comment by lashahub** [comment]
Thanks for looping me in.

AF-Next should not be added as a separate vLLM architecture anymore. The current HF checkpoints use the existing MusicFlamingo architecture (`model_type=musicflamingo`, `MusicFlamingoForConditionalGeneration`).

I’ll wait for this PR to settle/land, then rebase #39011 on top and re-scope it to AF-Next checkpoint coverage through MusicFlamingo only. That should avoid duplicating the Flamingo processor/RoTE fixes here.

**Review by hmellor — APPROVED** [review]
This LGTM, I'd like @eustlb to have one last look at the audio code before we press the big green button though

**Comment by AndreasKaratzas** [comment]
Hi @eustlb, could you please confirm?

**Comment by AndreasKaratzas** [comment]
cc @eustlb PTAL

**Review by eustlb — APPROVED** [review]
LGTM on the calls to transformers' processors for the audio part (audioflamingo + musicflaming) 👍
